Automattic/wp-calypso
# Plans: Remove `getDiscountedRawPrice` selector from Redux state

Part of addressing https://github.com/Automattic/wp-calypso/issues/86638

## Proposed Changes

Following up from https://github.com/Automattic/wp-calypso/pull/89436:
- we remove the `getDiscountedRawPrice` selector from Redux state as no longer used, along with any tests (no real need to port these across)

## Testing Instructions



* Ideally TC should catch any failures
* Confirm no lingering references to the selector that may go unnoticed in the build

## Pre-merge Checklist
